WordPress/Classes/ViewRelated/Reader/Sorting/ReaderSortingOptionButton.swift

```diff
@@ -0,0 +1,124 @@
+import UIKit
+
+extension ReaderSortingOption {
+
+    var localizedDescription: String? {
+        // TODO: check if strings are localized
+        switch self {
+        case .date:
+            return NSLocalizedString("Recent", comment: "Sorting option description")
+        case .popularity:
+            return NSLocalizedString("Popular", comment: "Sorting option description")
+        case .noSorting:
+            return nil
+        }
+    }
+
+    var image: UIImage? {
+        switch self {
+        case .date:
+            // TODO: use proper icon
+            return .gridicon(.calendar)
+        case .popularity:
+            // TODO: use proper icon
+            return .gridicon(.lineGraph)
+        case .noSorting:
+            return nil
+        }
+    }
+}
+
+class ReaderSortingOptionButton: UIControl {
+    var sourceView: UIView {
+        return chevronView
+    }
+
+    private lazy var iconView: UIImageView = {
+        let view = UIImageView()
+        view.tintColor = UIColor(light: UIColor(hexString: "4D4D4D"), dark: UIColor(hexString: "BFBFBF"))
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private lazy var label: UILabel = {
+        let view = UILabel()
+        view.textColor = UIColor(light: .black, dark: .white)
+        view.font = WPFontManager.systemSemiBoldFont(ofSize: 15.0)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private lazy var chevronView: UIImageView = {
+        let view = UIImageView()
+        view.image = .gridicon(.chevronDown)
+        view.tintColor = UIColor(light: UIColor(hexString: "4D4D4D"), dark: UIColor(hexString: "BFBFBF"))
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    var sortingOption: ReaderSortingOption? {
+        didSet {
+            label.text = sortingOption?.localizedDescription
+            iconView.image = sortingOption?.image
+        }
+    }
+
+    // TODO: check how should it behave when touched/highlighted
+    override var isHighlighted: Bool {
```

Comment:
would you mind checking if we already have a type in the app that already implements default highlighting behaviors? I'm specifically thinking of the filter button in the "Following" tab

Reply:
Filter button sets specific colors for normal, selected and highlighted state since it is a subclass of `UIButton`.
I found `AnimatedControl` but that looks a bit too much for this simple case.
Since my control has 2 icons and a label I went with subclassing `UIControl` and with basic override of `isHighlighted`. That is why I also added `TODO` to check how should I handle highlighting

Reply:
> Filter button sets specific colors for normal, selected and highlighted state since it is a subclass of UIButton.
Do you think it would be possible to use that same subclass then, since it's already there?

Reply:
Sounds good. Can you please add the alpha values in the `Constants` as well?

Reply:
Sure! For future "proper" solution we can go with a similar patter as `UIButton` has and have special colors/style for highlighted, disabled and selected state which would be applied depending in what state this control is.

Reply:
If/when we have the styles for those states, we can update it to use them